src/gleam/string.gleam

```diff
@@ -207,6 +206,24 @@ pub fn append(to first: String, suffix second: String) -> String {
   |> iodata.to_string
 }
 
+/// Repeat a string `n` times.
+///
+/// ## Examples
+/// ```gleam
+/// repeat("ha", times: 3) == "hahaha"
+/// ```
+///
+fn repeat_help(chunk: String, result: String, repeats: Int) -> String {
+  case repeats <= 0 {
+     True -> result
+     False -> repeat_help(chunk, append(result, chunk), repeats - 1)
```

Comment:
I think we can optimise this by building a list of strings can then calling `list.concat` on it, that way we don't need to copy the accumulator string each time we append a new piece.